### ISSUE TITLE
error code -10 on macos 10.12.4 during nanopore hybrid assembly

### ISSUE THREAD

**Issue by rec3141** [ablab/spades#25]
here's the end of the log file before it crashes, any help would be much appreciated!

```
  0:41:02.267     5G / 7G    INFO    General                 (kmer_splitters.hpp        : 225)   Processed 159541 edges
  0:41:06.135     5G / 7G    INFO    General                 (kmer_splitters.hpp        : 225)   Processed 320173 edges
  0:41:09.491     5G / 7G    INFO    General                 (kmer_splitters.hpp        : 225)   Processed 457944 edges
  0:41:09.491     5G / 7G    INFO    General                 (kmer_splitters.hpp        : 230)   Used 457944 sequences.
  0:41:09.537     4G / 7G    INFO    General                 (kmer_index_builder.hpp    : 252)   Starting k-mer counting.
  0:41:23.481     4G / 7G    INFO    General                 (kmer_index_builder.hpp    : 258)   K-mer counting done. There are 52636108 kmers in total.
  0:41:23.481     4G / 7G    INFO    General                 (kmer_index_builder.hpp    : 260)   Merging temporary buckets.
  0:41:24.686     4G / 7G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 437)   Building perfect hash indices
  0:41:41.979     4G / 7G    INFO    General                 (kmer_index_builder.hpp    : 276)   Merging final buckets.
  0:41:43.745     4G / 7G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 483)   Index built. Total 17197336 bytes occupied (2.61377 bits per kmer).
  0:41:44.566     6G / 7G    INFO    General                 (perfect_hash_map.hpp      : 185)   Arranging kmers in hash map order
  0:42:01.700     6G / 7G    INFO    General                 (perfect_hash_map.hpp      : 197)   Done. Total swaps: 52635845
  0:42:01.701     6G / 7G    INFO    General                 (edge_index_builders.hpp   :  27)   Collecting k-mer coverage information from graph, this takes a while.
  0:42:15.050    11G / 11G   INFO   PacIndex                 (pac_index.hpp             :  85)   Index constructed
  0:42:19.995    12G / 12G   INFO    General                 (hybrid_aligning.cpp       : 315)   Prepared batch 0 of 50000 reads.

== Error ==  system call for: "['/Volumes/work/cryomics/apps/SPAdes-3.10.1-Darwin/bin/spades', '/Volumes/work/cryomics/seq_data/Sphaeroforma_sirkka_B5/minION_B5_20170505_great/spades_hybrid_all/K55/configs/config.info', '/Volumes/work/cryomics/seq_data/Sphaeroforma_sirkka_B5/minION_B5_20170505_great/spades_hybrid_all/K55/configs/careful_mode.info']" finished abnormally, err code: -10
```
[params.txt](https://github.com/ablab/spades/files/1099409/params.txt)
[spades.log.txt](https://github.com/ablab/spades/files/1099408/spades.log.txt)

**Comment by rec3141** [comment]
a sample of the nanopore reads
[nanopore_test.txt](https://github.com/ablab/spades/files/1099410/nanopore_test.txt)

**Comment by asl** [comment]
Is there any possibility for you to share your data with us so we could reproduce and fix an issue?

**Comment by rec3141** [comment]
The data are here
https://drive.google.com/open?id=0B4JUBQuE6tBCSnRtV1F0RmlnRzQ

The assembly completes when using a subset of the HiSeq reads, up to about 10M paired end reads

8-core MacPro with 128Gb RAM

**Comment by asl** [comment]
Thanks, we downloaded all the files. Which nanopore reads you used? B5_all or B5_qs10? Your params.txt indicates that you used "B5.correctedReads_rename.fasta.gz"

**Comment by rec3141** [comment]
I tried it with multiple sets of nanopore reads in several different
formats (fq, fa, fq.gz) and it failed in each, including these.

**Comment by rec3141** [comment]
any ideas?

**Comment by asl** [comment]
We introduced some changes and the problem should be fixed in SPAdes 3.11

**Comment by asl** [comment]
Fixed in 3.11
